codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/CloseResponseMiddlewareGenerator.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ *
+ */
+
+package software.amazon.smithy.go.codegen.integration;
+
+import java.util.List;
+import java.util.Optional;
+import software.amazon.smithy.go.codegen.SmithyGoDependency;
+import software.amazon.smithy.go.codegen.SymbolUtils;
+import software.amazon.smithy.model.knowledge.HttpBinding;
+import software.amazon.smithy.model.knowledge.HttpBindingIndex;
+import software.amazon.smithy.model.shapes.MemberShape;
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.model.traits.StreamingTrait;
+import software.amazon.smithy.utils.ListUtils;
+
+public class CloseResponseMiddlewareGenerator implements GoIntegration {
+    /**
+     * Gets the sort order of the customization from -128 to 127, with lowest
+     * executed first.
+     *
+     * @return Returns the sort order, defaults to -40.
+     */
+    @Override
+    public byte getOrder() {
+        return 127;
```

Comment:
Are we sure we won't need any customization after response body is closed?

Reply:
Yeah we absolutely could, but this can be updated to reflect relevant ordering.

I picked 127 (last) so that it can put it self as the very first middleware in the deserializer step. Which will make it the last one to be invoked during deserialization of the response.

Reply:
changes this to a utility that is added by the SDK's protocols.